layouts/shortcodes/card.html

```diff
@@ -1,49 +1,44 @@
-<div class="td-card card mb-4">
-  {{ with $.Get "header" }}
-    <div class="card-header">
-      {{ if eq $.Page.File.Ext "md" }}
-        {{ $.Get "header" | markdownify }}
-      {{ else }}
-        {{ $.Get "header" | htmlUnescape | safeHTML }}
-      {{ end }}
+{{/* Check parameter type */ -}}
+{{ with .Get "code" -}}
+  {{ if ne ( printf "%T" . ) "bool" -}}
+    {{ errorf "shortcode card: parameter 'code' must be either true or false" -}}
+  {{ end -}}
+{{ end -}}
```

Comment:
Is the type check really necessary? Why not interpret the value simply as truthy / falsy?

Reply:
This type check is here for a good reason: the correct syntax is `<card code=true>`. If you write `<card code="true">`, an error ist thrown. What happens if you omit the check? See shortcode `readfile` which has no type check (yet!?):

https://github.com/google/docsy/blob/3eb2cd0cfb26521d94cc8503e0092cc4a5c40cc2/layouts/shortcodes/readfile.html#L31

Users are expected to write `<readfile code="true">`, if they write `<readfile code=true>`, nothing happens. This will be surprising for users and I doubt that all users immediately see what they did wrong (giving parameter value `true` instead of `"true"`).

Does that make clear why I added the type check? Does this answer your question and resolve the issue?

**Additional note**: IMHO shortcode `readfile` does things wrong here: it treats parameter `code` as string and tests for string equality while the parameter `code` has boolean character and should be treated like that in the shortcode.

Reply:
Yes, that makes sense. In that case, the error message could be clearer, stating that a boolean value is expected, but another type was given. E.g.,

> {{ errorf "shortcode card: boolean value expected for parameter 'code', but got type %s" ( printf "%T" . ) -}}

You can even extract the value of `( printf "%T" . )` earlier and reuse it in both places.

Reply:
Improved error message as requested.